src/librustdoc/html/sources.rs

```diff
@@ -99,16 +99,15 @@ impl<'a> SourceCollector<'a> {
             href.push('/');
         });
         self.scx.ensure_dir(&cur)?;
-        let mut fname = p.file_name().expect("source has no filename").to_os_string();
-        fname.push(".html");
+
+        let src_fname =
+            String::from(p.file_name().expect("source has no filename").to_string_lossy());
+        let fname = OsString::from(src_fname.clone() + ".html");
```

Comment:
This clones twice: once with `.clone()` and once for the `+`. Can you remove `.clone()`?
```suggestion
        let fname = OsString::from(src_fname + ".html");
```